// jdoodle-api/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var https_1 = __importDefault(require("https"));
var run = function (language, script) {
    return new Promise(function (resolve, reject) {
        var req = https_1.default.request({
            method: "POST",
            hostname: "www.jdoodle.com",
            port: 443,
            path: "/engine/execute",
            headers: {
                "Content-Type": "application/json",
                "x-requested-with": "XMLHttpRequest",
                referer: "https://www.jdoodle.com",
            },
        }, function (res) {
            res.setEncoding("utf8");
            var responseBody = "";
            res.on("data", function (chunk) {
                responseBody += chunk;
            });
            res.on("end", function () {
                var response = JSON.parse(responseBody);
                resolve(response);
            });
        });
        req.on("error", reject);
        req.write(JSON.stringify({
            language: language.languageCode,
            versionIndex: language.versionIndex,
            script: script,
        }));
        req.end();
    });
};
exports.default = run;
module.exports = run;
